tachyon/base/sort_benchmark.cc

```diff
@@ -0,0 +1,259 @@
+}
+
+template <SortMethod kSortMethod>
+void BM_StableSortPartiallySortedData(benchmark::State& state) {
+  std::vector<uint64_t> data = GetPartiallySortedData(state.range(0));
+  std::vector<uint64_t> data2 = data;
+  for (auto _ : state) {
+    if constexpr (kSortMethod == SortMethod::kPdq) {
+      base::UnstableSort(data2.begin(), data2.end());
+    } else if constexpr (kSortMethod == SortMethod::kPowersort) {
+      base::StableSort(data2.begin(), data2.end());
+    } else if constexpr (kSortMethod == SortMethod::kStdStableSort) {
+      std::stable_sort(data2.begin(), data2.end());
+    } else if constexpr (kSortMethod == SortMethod::kStdSort) {
+      std::sort(data2.begin(), data2.end());
+    }
+    data2 = data;
+  }
+  benchmark::DoNotOptimize(data);
+  benchmark::DoNotOptimize(data2);
+}
+
+BENCHMARK_TEMPLATE(BM_StableSortRandomData, SortMethod::kPdq)
+    ->RangeMultiplier(2)
+    ->Range(1 << 5, 1 << 20);
+BENCHMARK_TEMPLATE(BM_StableSortRandomData, SortMethod::kStdSort)
+    ->RangeMultiplier(2)
+    ->Range(1 << 5, 1 << 20);
+BENCHMARK_TEMPLATE(BM_StableSortRandomData, SortMethod::kPowersort)
+    ->RangeMultiplier(2)
+    ->Range(1 << 5, 1 << 20);
+BENCHMARK_TEMPLATE(BM_StableSortRandomData, SortMethod::kStdStableSort)
+    ->RangeMultiplier(2)
+    ->Range(1 << 5, 1 << 20);
+
+BENCHMARK_TEMPLATE(BM_StableSortPartiallySortedData, SortMethod::kPdq)
+    ->RangeMultiplier(2)
+    ->Range(1 << 5, 1 << 20);
+BENCHMARK_TEMPLATE(BM_StableSortPartiallySortedData, SortMethod::kStdSort)
+    ->RangeMultiplier(2)
+    ->Range(1 << 5, 1 << 20);
+BENCHMARK_TEMPLATE(BM_StableSortPartiallySortedData, SortMethod::kPowersort)
+    ->RangeMultiplier(2)
+    ->Range(1 << 5, 1 << 20);
+BENCHMARK_TEMPLATE(BM_StableSortPartiallySortedData, SortMethod::kStdStableSort)
+    ->RangeMultiplier(2)
+    ->Range(1 << 5, 1 << 20);
+
+}  // namespace tachyon::math
+
+// clang-format off
+// Executing tests from //tachyon/base:sort_benchmark
+// -----------------------------------------------------------------------------
+// 2024-08-05T09:26:02+00:00
+// Running /home/chokobole/.cache/bazel/_bazel_chokobole/234690e3562329d13f7f07caac03dae4/execroot/kroma_network_tachyon/bazel-out/k8-opt/bin/tachyon/base/sort_benchmark.runfiles/kroma_network_tachyon/tachyon/base/sort_benchmark
+// Run on (32 X 5499.96 MHz CPU s)
+// CPU Caches:
+//   L1 Data 48 KiB (x16)
+//   L1 Instruction 32 KiB (x16)
+//   L2 Unified 2048 KiB (x16)
+//   L3 Unified 36864 KiB (x1)
+// Load Average: 2.58, 2.64, 2.46
+// ---------------------------------------------------------------------------------------------------------------
+// Benchmark                                                                     Time             CPU   Iterations
+// ---------------------------------------------------------------------------------------------------------------
+// BM_StableSortRandomData<SortMethod::kPdq>/32                               64.5 ns         64.5 ns     10927050
```

Comment:
In the benchmark results, it is indicated as if pdqsort is a stable sort, which could cause confusion.